src/main/java/bloodnet/logic/commands/EditDonationCommand.java

```
@@ -116,15 +119,15 @@ private DonationRecord getDonationRecordToEdit(Model model) throws CommandExcept
      */
     private Person getPersonToEditRecordFor(Model model, DonationRecord donationRecord) throws CommandException {
         requireNonNull(model);
+        requireNonNull(donationRecord);
         List<Person> personList = model.getFilteredPersonList();
```

Correct me if I'm wrong, but I believe the correct way to fix issue #142 is to change line 123 to `List<Person> personList = model.getBloodNet().getPersonList()`. This is so that this `EditDonationCommand` will try to find the Person object corresponding to `donationRecord.getPersonId()` in the FULL person list, instead of in the filtered person list.

Throwing a `CommandException` does not fix the problem. If a person corresponding to the donation record the user is trying to edit is not found, indeed a `CommandException` is thrown here, instead of a `NullPointerException` being thrown by the calling method (that is, .execute()).

Although the error is indeed displayed to the user nicely now, it does not fix the underlying issue that the user should be able to edit the donation record, even if the person is not shown in the donor list.

---

I agree. I was gonna mention this too HAHA

---

Great catch! I did that to be more consistent with deletedonation, but I can definitely edit this.

I guess the original idea was that findeligible would only be linked to the donation records from the filtered persons.

---

How about for adddonation or finddonations? If you were to filter, then how would you know which person it is for? 

Can mention in the enhancements though.